File: youtubejextractor/src/main/java/com/github/kotvertolet/youtubejextractor/models/test/VideoDetails.java
package com.github.kotvertolet.youtubejextractor.models.test;

import java.util.List;
import com.google.gson.annotations.SerializedName;

public class VideoDetails{

	@SerializedName("isOwnerViewing")
	private boolean isOwnerViewing;

	@SerializedName("thumbnail")
	private Thumbnail thumbnail;

	@SerializedName("isLiveContent")
	private boolean isLiveContent;

	@SerializedName("keywords")
	private List<String> keywords;

	@SerializedName("author")
	private String author;

	@SerializedName("lengthSeconds")
	private String lengthSeconds;

	@SerializedName("videoId")
	private String videoId;

	@SerializedName("shortDescription")
	private String shortDescription;

	@SerializedName("isPrivate")
	private boolean isPrivate;

	@SerializedName("title")
	private String title;

	@SerializedName("isCrawlable")
	private boolean isCrawlable;

	@SerializedName("averageRating")
	private double averageRating;

	@SerializedName("isUnpluggedCorpus")
	private boolean isUnpluggedCorpus;

	@SerializedName("allowRatings")
	private boolean allowRatings;

	@SerializedName("viewCount")
	private String viewCount;

	@SerializedName("channelId")
	private String channelId;

	public void setIsOwnerViewing(boolean isOwnerViewing){
		this.isOwnerViewing = isOwnerViewing;
	}

	public boolean isIsOwnerViewing(){
		return isOwnerViewing;
	}

	public void setThumbnail(Thumbnail thumbnail){
		this.thumbnail = thumbnail;
	}

	public Thumbnail getThumbnail(){
		return thumbnail;
	}

	public void setIsLiveContent(boolean isLiveContent){
		this.isLiveContent = isLiveContent;
	}

	public boolean isIsLiveContent(){
		return isLiveContent;
	}

	public void setKeywords(List<String> keywords){
		this.keywords = keywords;
	}

	public List<String> getKeywords(){
		return keywords;
	}

	public void setAuthor(String author){
		this.author = author;
	}

	public String getAuthor(){
		return author;
	}

	public void setLengthSeconds(String lengthSeconds){
		this.lengthSeconds = lengthSeconds;
	}

	public String getLengthSeconds(){
		return lengthSeconds;
	}

	public void setVideoId(String videoId){
		this.videoId = videoId;
	}

	public String getVideoId(){
		return videoId;
	}

	public void setShortDescription(String shortDescription){
		this.shortDescription = shortDescription;
	}

	public String getShortDescription(){
		return shortDescription;
	}

	public void setIsPrivate(boolean isPrivate){
		this.isPrivate = isPrivate;
	}

	public boolean isIsPrivate(){
		return isPrivate;
	}

	public void setTitle(String title){
		this.title = title;
	}

	public String getTitle(){
		return title;
	}

	public void setIsCrawlable(boolean isCrawlable){
		this.isCrawlable = isCrawlable;
	}

	public boolean isIsCrawlable(){
		return isCrawlable;
	}

	public void setAverageRating(double averageRating){
		this.averageRating = averageRating;
	}

	public double getAverageRating(){
		return averageRating;
	}

	public void setIsUnpluggedCorpus(boolean isUnpluggedCorpus){
		this.isUnpluggedCorpus = isUnpluggedCorpus;
	}

	public boolean isIsUnpluggedCorpus(){
		return isUnpluggedCorpus;
	}

	public void setAllowRatings(boolean allowRatings){
		this.allowRatings = allowRatings;
	}

	public boolean isAllowRatings(){
		return allowRatings;
	}

	public void setViewCount(String viewCount){
		this.viewCount = viewCount;
	}

	public String getViewCount(){
		return viewCount;
	}

	public void setChannelId(String channelId){
		this.channelId = channelId;
	}

	public String getChannelId(){
		return channelId;
	}
}

File: youtubejextractor/src/main/java/com/github/kotvertolet/youtubejextractor/models/newModels/VideoDetails.kt
package com.github.kotvertolet.youtubejextractor.models.newModels

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize
import java.io.Serializable

@Parcelize
class VideoDetails : Parcelable, Serializable {
    @SerializedName("isOwnerViewing")
    var isIsOwnerViewing = false
        private set

    @SerializedName("thumbnail")
    var thumbnail: Thumbnail? = null

    @SerializedName("isLiveContent")
    var isLiveContent = false

    @SerializedName("keywords")
    var keywords: List<String>? = null

    @SerializedName("author")
    var author: String? = null

    @SerializedName("lengthSeconds")
    var lengthSeconds: String? = null

    @SerializedName("videoId")
    var videoId: String? = null

    @SerializedName("shortDescription")
    var shortDescription: String? = null

    @SerializedName("isPrivate")
    var isIsPrivate = false
        private set

    @SerializedName("title")
    var title: String? = null

    @SerializedName("isCrawlable")
    var isIsCrawlable = false
        private set

    @SerializedName("averageRating")
    var averageRating = 0.0

    @SerializedName("isUnpluggedCorpus")
    var isIsUnpluggedCorpus = false
        private set

    @SerializedName("allowRatings")
    var isAllowRatings = false

    @SerializedName("viewCount")
    var viewCount: String? = null

    @SerializedName("channelId")
    var channelId: String? = null
    fun setIsOwnerViewing(isOwnerViewing: Boolean) {
        isIsOwnerViewing = isOwnerViewing
    }

    fun setIsPrivate(isPrivate: Boolean) {
        isIsPrivate = isPrivate
    }

    fun setIsCrawlable(isCrawlable: Boolean) {
        isIsCrawlable = isCrawlable
    }

    fun setIsUnpluggedCorpus(isUnpluggedCorpus: Boolean) {
        isIsUnpluggedCorpus = isUnpluggedCorpus
    }

    override fun toString(): String {
        return "VideoDetails{" +
                "isOwnerViewing = '" + isIsOwnerViewing + '\'' +
                ",thumbnail = '" + thumbnail + '\'' +
                ",isLiveContent = '" + isLiveContent + '\'' +
                ",keywords = '" + keywords + '\'' +
                ",author = '" + author + '\'' +
                ",lengthSeconds = '" + lengthSeconds + '\'' +
                ",videoId = '" + videoId + '\'' +
                ",shortDescription = '" + shortDescription + '\'' +
                ",isPrivate = '" + isIsPrivate + '\'' +
                ",title = '" + title + '\'' +
                ",isCrawlable = '" + isIsCrawlable + '\'' +
                ",averageRating = '" + averageRating + '\'' +
                ",isUnpluggedCorpus = '" + isIsUnpluggedCorpus + '\'' +
                ",allowRatings = '" + isAllowRatings + '\'' +
                ",viewCount = '" + viewCount + '\'' +
                ",channelId = '" + channelId + '\'' +
                "}"
    }
}